pkg/deploy/util/util.go

```diff
@@ -429,6 +433,42 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	return relevantDeployments
 }
 
+// WaitForRunningDeployerPod waits a given period of time until the deployer pod
+// for given replication controller is not running.
+func WaitForRunningDeployerPod(podClient kclient.PodsNamespacer, rc *api.ReplicationController, timeout time.Duration) (bool, error) {
+	podName := DeployerPodNameForDeployment(rc.Name)
+	if len(podName) == 0 {
+		return false, nil
+	}
+	pod, err := podClient.Pods(rc.Namespace).Get(podName)
+	if err != nil || (pod.Status.Phase == api.PodRunning) {
```

Comment:
There is [IsPodActive](https://github.com/kubernetes/kubernetes/blob/1e7545393f556529aa753528f1368b50f4259411/pkg/controller/controller_utils.go#L699) that covers the terminal states.